janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphVertexProperty.java

```diff
@@ -45,4 +47,16 @@ default PropertyKey propertyKey() {
         return (PropertyKey)getType();
     }
 
+    static Consumer<JanusGraphVertexProperty> getRemover(VertexProperty.Cardinality cardinality, Object value) {
+        if (cardinality == VertexProperty.Cardinality.single) {
+            return JanusGraphElement::remove;
```

Comment:
I don't know much about how lambda are compiled but if it is changed to `new Consumer<JanusGraphVertexProperty> {....}` shouldn't this consumer be static? I don't think we need to create a new instance of this consumer because it doesn't reuse any other parameters.
I.e., I think it would be better to create a static consumer like:
```
private static Consumer<JanusGraphVertexProperty> SINGLE_PROPERTY_REMOVER_CONSUMER = JanusGraphElement::remove; 
```
and reuse `SINGLE_PROPERTY_REMOVER_CONSUMER` here. That said, if you know that `JanusGraphElement::remove` will be compiled to static consumer, I'm totally good with it.

Reply:
Looks like it is compiled to static consumer:

```java
boolean sameObject = false;
Consumer<JanusGraphVertexProperty> c1 = null;
for (int i = 0; i < 2; i++) {
    Consumer<JanusGraphVertexProperty> c2 = new Consumer<JanusGraphVertexProperty>() {
        @Override
        public void accept(JanusGraphVertexProperty p) {
            p.remove();
        }
    };
    if (c1 == null) c1 = c2;
    else sameObject = c1 == c2; // false
}
```

when using method reference or lambda function, result is different:

```java
boolean sameObject = false;
Consumer<JanusGraphVertexProperty> c1 = null;
for (int i = 0; i < 2; i++) {
    Consumer<JanusGraphVertexProperty> c2 = JanusGraphElement::remove;
    if (c1 == null) c1 = c2;
    else sameObject = c1 == c2; // true
}
```

Example modified from https://stackoverflow.com/questions/23983832/is-method-reference-caching-a-good-idea-in-java-8/23991339#23991339

Reply:
Thank you for checking that!